### PR TITLE
added X509 cert signature, option to add timestamp at end of security header, PSR4

### PR DESCRIPTION
**Pull request by bendubuisson**

### CODE REVIEW

**Comment by bendubuisson** [comment]
@robrichards If you got some time, it would be great to get this merged and released? Love your work 👍

File: composer.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
         }
     },
     "require": {
-        "php": ">= 5.3",
```
Comment (robrichards) [inline composer.json]: Is there a reason that this has been bumped up?

File: src/WSSESoap.php [hunk 2]
```diff
@@ -281,30 +403,32 @@ public function signSoapDoc($objKey, $options = null)
...
+            $insertTop = (bool)$options['insertBefore'];
         }
         $objDSig->appendSignature($this->secNode, $insertTop);
 
         /* New suff */
         if (is_array($options)) {
             if (!empty($options['KeyInfo'])) {
                 if (!empty($options['KeyInfo']['X509SubjectKeyIdentifier'])) {
-                    $sigNode = $this->secNode->firstChild->nextSibling;
+                    $sigNode = $this->secNode->firstChild;
```
Comment (robrichards) [inline src/WSSESoap.php]: Why needed?

**Review by robrichards — COMMENTED** [review]
Few questions

File: composer.json (hunk 1 above)
Comment (bendubuisson) [reply composer.json]: PHP 5.* is not supported anymore https://www.php.net/supported-versions.php

File: src/WSSESoap.php (hunk 2 above)
Comment (bendubuisson) [reply src/WSSESoap.php]: To be honest I do not remember as I made those changes over a year ago, I will have to do some digging and get back to you. Thanks for your time.

File: composer.json (hunk 1 above)
Comment (robrichards) [reply composer.json]: There are still people using it even tho its no longer supported and unless there is something in the library that requires a newer version I am trying to keep it compatible with older versions

Comment (robrichards) [reply composer.json]: BTW I could see bumping this to 5.4 if needed as the dependent xmlseclibs library requires at least that version.

**Pull request closed without merging**